tools/flaky_test_monitor/level2/process_summary2_results.go

```diff
@@ -0,0 +1,139 @@
+	err = os.Mkdir(failuresDir, 0755)
+	helpers.AssertErrNil(err, "error creating failures directory")
+
+	testSummary2 := helpers.TestSummary2{}
+	testSummary2.TestResults = make(map[string]*helpers.TestResultSummary)
+
+	// go through all level 1 summaries in a folder to create level 2 summary
+	for i := 0; i < len(dirEntries); i++ {
+		// read in each level 1 summary
+		var level1TestRun helpers.TestRun
+
+		level1JsonBytes, err := os.ReadFile(level1Directory + dirEntries[i].Name())
+		helpers.AssertErrNil(err, "error reading level 1 json")
+
+		err = json.Unmarshal(level1JsonBytes, &level1TestRun)
+		helpers.AssertErrNil(err, "error unmarshalling level 1 test run")
+
+		// go through each level 1 summary and update level 2 summary
+		for _, packageResult := range level1TestRun.PackageResults {
+			for _, testResult := range packageResult.Tests {
+				// check if already started collecting summary for this test
+				mapKey := testResult.Package + "/" + testResult.Test
+				testResultSummary, testResultSummaryExists := testSummary2.TestResults[mapKey]
+
+				// this test doesn't have a summary so create it
+				// no need to specify other fields explicitly - default values will suffice
+				if !testResultSummaryExists {
+					testResultSummary = &helpers.TestResultSummary{
+						Test:    testResult.Test,
+						Package: testResult.Package,
+					}
+				}
+
+				// keep track of each duration so can later calculate average duration
+				testResultSummary.Durations = append(testResultSummary.Durations, testResult.Elapsed)
+
+				// increment # of passes, fails, skips or no results for this test
+				testResultSummary.Runs++
+				switch testResult.Result {
+				case "pass":
+					testResultSummary.Passed++
+				case "fail":
+					testResultSummary.Failed++
+					saveFailureMessage(testResult)
+				case "skip":
+					testResultSummary.Skipped++
+
+					// don't count skip as a run
+					testResultSummary.Runs--
+
+					// truncate last duration - don't count durations of skips
+					testResultSummary.Durations = testResultSummary.Durations[:len(testResultSummary.Durations)-1]
+				case "":
+					testResultSummary.NoResult++
+				default:
+					panic(fmt.Sprintf("unexpected test result: %s", testResult.Result))
+				}
+
+				testSummary2.TestResults[mapKey] = testResultSummary
+			}
+		}
+	}
+
+	// calculate averages and other calculations that can only be completed after all test runs have been read
+	postProcessTestSummary2(testSummary2)
+	return testSummary2
+}
+
+func saveFailureMessage(testResult helpers.TestResult) {
+	if testResult.Result != "fail" {
+		panic(fmt.Sprintf("unexpected test result: " + testResult.Result))
+	}
+
+	// sub-directory names should be the same - each sub directory corresponds to a failed test name
+	// there could already be previous failures for this test, so it's important to check if failed
+	// test folder exists
+	if !helpers.FolderExists(failuresDir + testResult.Test) {
```

Comment:
Do we want to include the package as well in the name? It's possible that two tests from different packages could have the same name.

Reply:
Done
https://github.com/onflow/flow-go/pull/1707/commits/d815858ca2d9e539bfaaf3a56c0f1e4e58973642